pyqir-generator/tests/test_external_functions.py

```diff
@@ -203,3 +203,74 @@ def test_wrong_number_of_args(self) -> None:
                 message = f"Expected {len(param_types)} arguments, got {len(args)}."
                 with self.assertRaisesRegex(ValueError, message):
                     mod.builder.call(f, args)
+
+    def test_variable(self) -> None:
+        mod = SimpleModule("test", 0, 0)
+        foo = mod.add_external_function("foo", types.Function([], types.INT))
+        bar = mod.add_external_function(
+            "bar", types.Function([types.INT], types.VOID))
+
+        x = mod.builder.call(foo, [])
+        mod.builder.call(bar, [x])
+
+        ir = mod.ir()
+        self.assertIn("%0 = call i64 @foo()", ir)
+        self.assertIn("call void @bar(i64 %0)", ir)
+
+    def test_variable_wrong_external_type(self) -> None:
+        mod = SimpleModule("test", 0, 0)
+        foo = mod.add_external_function("foo", types.Function([], types.INT))
+        bar = mod.add_external_function(
+            "bar", types.Function([types.QUBIT], types.VOID))
+
+        x = mod.builder.call(foo, [])
+        mod.builder.call(bar, [x])
+
+        with self.assertRaisesRegex(OSError, "Call parameter type does not match function signature!"):
```

Comment:
To be honest I'm not sure where the OSError is coming from specifically, if we wanted to change it to another exception type. It seems like it's consistent with other LLVM errors that raise OSError, though.